src/components/playback/playbackmanager.js

```diff
@@ -1789,42 +1790,73 @@ class PlaybackManager {
             });
         }
 
-        function translateItemsForPlayback(items, options) {
+        async function translateItemsForPlayback(items, options) {
             if (!items.length) return Promise.resolve([]);
 
-            if (items.length > 1 && options && options.ids) {
+            sortItemsIfNeeded(items, options);
+
+            const firstItem = items[0];
+            const serverId = firstItem.ServerId;
+            const queryOptions = options.queryOptions || {};
+
+            const promise = getPlaybackPromise(firstItem, serverId, options, queryOptions, items);
+
+            if (promise) {
+                const result = await promise;
+                return result ? result.Items : items;
+            } else {
+                return Promise.resolve(items);
+            }
+        }
+
+        function sortItemsIfNeeded(items, options) {
+            if (items.length > 1 && options?.ids) {
                 // Use the original request id array for sorting the result in the proper order
                 items.sort(function (a, b) {
                     return options.ids.indexOf(a.Id) - options.ids.indexOf(b.Id);
                 });
             }
+        }
 
-            const firstItem = items[0];
-            let promise;
-
-            const serverId = firstItem.ServerId;
-
-            const queryOptions = options.queryOptions || {};
+        function getPlaybackPromise(firstItem, serverId, options, queryOptions, items) {
+            const nonItemTypePromise = getNonItemTypePromise(firstItem, serverId, options, queryOptions);
+            if (nonItemTypePromise) {
+                return nonItemTypePromise;
+            }
+
+            switch (firstItem.Type) {
+                case 'Program':
+                    return getItemsForPlayback(serverId, { Ids: firstItem.ChannelId });
+                case 'Playlist':
+                    return getItemsForPlayback(serverId, { ParentId: firstItem.Id, SortBy: options.shuffle ? 'Random' : null });
+                case 'MusicArtist':
+                    return getItemsForPlayback(serverId, mergePlaybackQueries({ ArtistIds: firstItem.Id, Filters: 'IsNotFolder', Recursive: true, SortBy: options.shuffle ? 'Random' : 'SortName', MediaTypes: 'Audio' }, queryOptions));
```

Comment:
I would keep the old formatting so there are fewer changes.
```suggestion
                    return getItemsForPlayback(serverId, mergePlaybackQueries({
                        ArtistIds: firstItem.Id,
                        Filters: 'IsNotFolder',
                        Recursive: true,
                        SortBy: options.shuffle ? 'Random' : 'SortName',
                        MediaTypes: 'Audio'
                    }, queryOptions));
```

Reply:
I am assuming by this you are referring to the line spacing for the params for each `getItemsForPlayback` function call in the switch block. I have updated that so it matches the previous formatting.

If you mean going back to the if/else-if chain, I think I would be curious to hear more of the logic behind why you feel that should change back? I'll still change it if you have strong opinions about it, but to me having the switch makes more sense since all but 3 conditions involved checking the exact same field. If my first statement is what you were asking for, disregard this then